src/client/Client.cc

```diff
@@ -2460,7 +2460,7 @@ void Client::_handle_full_flag(int64_t pool)
   {
     Inode *inode = i->second;
     if (inode->oset.dirty_or_tx
-        && (pool == -1 || inode->layout.pool_id == pool)) {
+        && inode->layout.pool_id == pool) {
```

Comment:
I think this means that if we use the octopus client to talk to a pre-luminous (?) cluster, which doesn't set per-pool full flags, we'll misbehave?

If that's what we want to do, we need to document it.  I'm a bit unsure we should drop this code, though.

Reply:
@liewegas i dropped this commit from this PR.